source/common/memory/stats.cc

```diff
@@ -40,6 +42,13 @@ uint64_t Stats::totalPageHeapUnmapped() {
   return value;
 }
 
+void Stats::dumpStatsToLog() {
+  static const int kBufferSize = 3 << 20;
```

Comment:
This is some pretty weird math! Nit: make this `constexpr` and call it `buffer_size`.